### PR TITLE
Cypress test update and related linting fixes

### PR DESCRIPTION
**Pull request by mengweieric**
### Description
- Update event analytics cypress
- Fix linting errors

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuali925 — APPROVED** [review]

File: common/types/explorer.ts [hunk 1]
```diff
@@ -380,7 +380,7 @@ export interface VisMeta {
 }
 
 export interface VisualizationState {
-  query: Query;
+  queryState: Query;
```
Comment (derek-ho) [inline common/types/explorer.ts:383]: any reason why this was changed?

Comment (mengweieric) [reply common/types/explorer.ts:383]: linting fix, there's a global 'query' within the context where inner query will shadow the global one.

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_fields.tsx [hunk 2]
```diff
@@ -55,10 +55,10 @@ export const DataConfigPanelFields = ({
     if (visType === VIS_CHART_TYPES.HeatMap)
       return name === AGGREGATIONS ? list.length >= 1 : list.length >= 2;
     // condition for line and scatter for dimensions section.
-    return name === GROUPBY && (list.length >= 1 || !isEmpty(time_field));
+    return name === GROUPBY && (list.length >= 1 || !isEmpty(timeField));
   };
 
-  const { time_field, unit, interval } = dimensionSpan;
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_fields.tsx:61]: Again why was this field name changed?

Comment (mengweieric) [reply public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_fields.tsx:61]: Same linting fix, rules only accept camelCase, CamelCase, or CAMEL_CASE

File: .cypress/utils/event_analytics/constants.js [hunk 3]
```diff
@@ -20,6 +20,7 @@ export const TEST_QUERIES = [
   },
   {
     query: 'source = opensearch_dashboards_sample_data_logs',
+    dateRangeDOM: YEAR_TO_DATE_DOM_ID,
```
Comment (derek-ho) [inline .cypress/utils/event_analytics/constants.js:23]: Question: how is this index timestamp determined - is it when you add the data? Will this break next year?

File: .cypress/integration/1_event_analytics.spec.js [hunk 4]
```diff
@@ -280,19 +229,14 @@ describe('Saves a query on explorer page', () => {
     cy.get('[data-test-subj="eventExplorer__querySaveName"]')
       .focus()
       .type(SAVE_QUERY2, { force: true });
-
-    cy.get('[data-test-subj="eventExplorer__querySaveConfirm"]').click();
-    cy.wait(delay * 2);
-    cy.get('.euiToastHeader__title').contains('successfully').should('exist');
-
+    cy.get('[data-test-subj="eventExplorer__querySaveConfirm"]').click({ force: true });
+    cy.get('.euiToastHeader__title').should('contain', 'successfully');
```
Comment (derek-ho) [inline .cypress/integration/1_event_analytics.spec.js:233]: is it really ok to get rid of the delay here? Won't the request take some time?

Comment (derek-ho) [reply .cypress/integration/1_event_analytics.spec.js:233]: Same concern in a few places

File: .cypress/integration/1_event_analytics.spec.js [hunk 5]
```diff
@@ -18,79 +18,32 @@ import {
...
+  VIS_TYPE_HBAR,
+  VIS_TYPE_VBAR,
+  VIS_TYPE_HEATMAP,
+  FIELD_HOST,
+  FIELD_AGENT
 } from '../utils/event_analytics/constants';
 import { supressResizeObserverIssue, COMMAND_TIMEOUT_LONG } from '../utils/constants';
 import { clearQuerySearchBoxText } from '../utils/event_analytics/helpers';
 
-const renderHistogramChart = () => {
```
Comment (derek-ho) [inline .cypress/integration/1_event_analytics.spec.js:31]: Why were these tests removed?

File: public/components/event_analytics/explorer/save_panel/__tests__/__snapshots__/save_panel.test.tsx.snap [hunk 6]
```diff
@@ -738,7 +738,7 @@ exports[`Saved query table component Renders saved query table 1`] = `
             onMouseOver={[Function]}
           >
             <EuiSwitch
-              checked={true}
+              checked={false}
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/save_panel/__tests__/__snapshots__/save_panel.test.tsx.snap:741]: was this behavior not tested properly - this seems like a pretty big snapshot change?

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx [hunk 7]
```diff
@@ -50,59 +42,36 @@ export const SavePanel = ({
...
-  setMetricMeasure,
-  setMetricLabel,
-  metricChecked,
+  isSaveAsMetricEnabled,
 }: ISavedPanelProps) => {
   const [options, setOptions] = useState([]);
   const [checked, setChecked] = useState(false);
-  const [measure, setMeasure] = useState('');
-  const [label, setLabel] = useState([]);
+  const [svpnlError, setSvpnlError] = useState(null);
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/save_panel/save_panel.tsx:50]: What are you doing this this value?

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx [hunk 8]
```diff
@@ -50,59 +42,36 @@ export const SavePanel = ({
...
-    setLabel(selectedLabels);
-    setMetricLabel(selectedLabels);
-  };
-
-  useEffect(() => {
-    if (metricChecked) {
-      setChecked(true);
-      setMeasure(metricMeasure);
-    }
-  }, [])
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/save_panel/save_panel.tsx:104]: is this what was causing snapshot change? Is this deletion intended?

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx [hunk 9]
```diff
@@ -50,59 +42,36 @@ export const SavePanel = ({
...
 
-  const onMeasureChange = (selectedMeasures: string) => {
-    setMeasure(selectedMeasures);
-    setMetricMeasure(selectedMeasures);
-  };
-
-  const onLabelChange = (selectedLabels: React.SetStateAction<never[]>) => {
-    setLabel(selectedLabels);
-    setMetricLabel(selectedLabels);
-  };
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/save_panel/save_panel.tsx:97]: these were just unused?

File: .cypress/integration/1_event_analytics.spec.js (hunk 5 above)
Comment (mengweieric) [reply .cypress/integration/1_event_analytics.spec.js:31]: We don't need these any more.

File: .cypress/utils/event_analytics/constants.js (hunk 3 above)
Comment (mengweieric) [reply .cypress/utils/event_analytics/constants.js:23]: timestamp is determined automatically in event explorer by looking for the first encountered date type in an index mappings. It will not break as long as the sample data is not changed.

File: public/components/common/search/search.tsx [hunk 10]
```diff
@@ -221,15 +217,12 @@ export const Search = (props: any) => {
...
                   curVisId={curVisId}
-                  spanValue={spanValue}
                   setSubType={setSubType}
-                  metricMeasure={metricMeasure}
-                  setMetricMeasure={setMetricMeasure}
-                  setMetricLabel={setMetricLabel}
-                  metricChecked={metricChecked}
+                  isSaveAsMetricEnabled={
+                    isEqual(curVisId, 'line') && tempQuery.match(PPL_SPAN_REGEX) !== null
+                  }
```
Comment (derek-ho) [inline public/components/common/search/search.tsx:225]: Can you explain what is this doing/why this change was made here? Any issue linking the bug here?

File: public/components/event_analytics/explorer/explorer.tsx [hunk 11]
```diff
@@ -1430,25 +1437,17 @@ export const Explorer = ({
...
   };
 
-  useEffect(() => {
-    if (isEqual(selectedContentTabId, TAB_CHART_ID)) {
-      const statsTokens = queryManager.queryParser().parse(tempQuery).getStats();
-      const updatedDataConfig = getDefaultVisConfig(statsTokens);
-      setSpanValue(!isEqual(typeof updatedDataConfig.span, 'undefined'));
-    }
-  }, [tempQuery, selectedContentTabId, curVisId]);
-
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/explorer.tsx:1451]: why removed?

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx (hunk 7 above)
Comment (mengweieric) [reply public/components/event_analytics/explorer/save_panel/save_panel.tsx:50]: the linting rule doesn't allow using console.xxx statement, so here I've just done a quick change to make error as a state (so later we can pipe error to maybe toast UI) rather than using console as this approach may not help on knowing the error.

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx [hunk 12]
```diff
@@ -146,18 +115,17 @@ export const SavePanel = ({
...
           <EuiFormRow display="columnCompressedSwitch">
-          <EuiToolTip
-          content="Only Time Series visualization and a query including stats/span can be saved as Metric" >
-            <EuiSwitch
-              showLabel={true}
-              label="Save as Metric"
-              checked={checked}
-              onChange={onToggleChange}
-              compressed
-              disabled = {!((isEqual(curVisId, 'line')) && spanValue)}
```
Comment (derek-ho) [inline public/components/event_analytics/explorer/save_panel/save_panel.tsx:157]: Never mind I see that this is passed down now, but wondering if matching the regex is doing the same as this block of code: 
`useEffect(() => {
    if (isEqual(selectedContentTabId, TAB_CHART_ID)) {
      const statsTokens = queryManager.queryParser().parse(tempQuery).getStats();
      const updatedDataConfig = getDefaultVisConfig(statsTokens);
      setSpanValue(!isEqual(typeof updatedDataConfig.span, 'undefined'));
    }
  }, [tempQuery, selectedContentTabId, curVisId]);`

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx (hunk 9 above)
Comment (mengweieric) [reply public/components/event_analytics/explorer/save_panel/save_panel.tsx:97]: For now I don't see these functions are used at all, linting is complaining about unused stuff so I was just doing minimal removals here to pass linting checking

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx [hunk 13]
```diff
@@ -113,8 +82,8 @@ export const SavePanel = ({
           <EuiFormRow helpText="Search existing dashboards or applications by name">
             <EuiComboBox
               placeholder="Select dashboards/applications"
-              onChange={(options) => {
-                handleOptionChange(options);
+              onChange={(daOptions) => {
```
[the remark this answers is not in the document]
Comment (mengweieric) [reply public/components/event_analytics/explorer/save_panel/save_panel.tsx:85]: Similar reason, 'options' is a global defined variable, have to rename to avoid shadowing and pass the linting checking

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx (hunk 8 above)
Comment (mengweieric) [reply public/components/event_analytics/explorer/save_panel/save_panel.tsx:104]: yes intended, I recall this should also relate to unused variable issues.

File: public/components/common/search/search.tsx (hunk 10 above)
Comment (mengweieric) [reply public/components/common/search/search.tsx:225]: This comes from linting autofix.

File: public/components/event_analytics/explorer/explorer.tsx (hunk 11 above)
Comment (mengweieric) [reply public/components/event_analytics/explorer/explorer.tsx:1451]: The way metric uses to determine disabled/enabled saving-as-metric, as far as I can see, only cares about if the query contains timespan on saving-panel open. This code piece does the work to record timespan but it triggers query manager parsing and setSpanValue on every query typing/changing which causes performance degradation.

File: public/components/event_analytics/explorer/save_panel/save_panel.tsx (hunk 12 above)
Comment (mengweieric) [reply public/components/event_analytics/explorer/save_panel/save_panel.tsx:157]: yes the quick change i made is to match a span in query.

File: .cypress/integration/1_event_analytics.spec.js (hunk 4 above)
Comment (mengweieric) [reply .cypress/integration/1_event_analytics.spec.js:233]: The wait here is like a hard wait, event request has returned successfully, it would still wait there. we don't necessary need to wait here for a specific amount of time. I think wait is something we need to almost always universally avoid in our test code, we should work on gradually refactor code to use other approaches.

**Review by derek-ho — APPROVED** [review]
LGTM

**Comment by ps48** [comment]
@mengweieric should we backport these changes to 2.x?
